### PR TITLE
[Feature #19538] Add "performance" warning category

### PR DESCRIPTION
**Pull request by nobu**
https://bugs.ruby-lang.org/issues/19538

### CODE REVIEW

**Comment by casperisfine** [comment]
Oh, I was working on it 😄

File: include/ruby/internal/error.h [hunk 1]
```diff
@@ -50,6 +50,9 @@ typedef enum {
     /** Warning is for experimental features. */
     RB_WARN_CATEGORY_EXPERIMENTAL,
 
+    /** Warning is for performance. */
+    RB_WARN_CATEGORY_PERFORMANCE,
```
Comment (paracycle) [inline include/ruby/internal/error.h:54]: Shouldn't this also be added to the `RB_WARN_CATEGORY_ALL_BITS` below?

Comment (casperisfine) [reply include/ruby/internal/error.h:54]: Nope, it was specifically asked that this category is not enabled on bare`-W`. `RB_WARN_CATEGORY_ALL_BITS` is what controls that.

**Comment by hsbt** [comment]
We merged https://github.com/ruby/ruby/pull/7708 instead of this.

**Pull request closed without merging**